lib/modules/onc_program/onc_standalone_public_launch_sequence.rb

```diff
@@ -0,0 +1,188 @@
+# frozen_string_literal: true
+
+require_relative './shared_onc_launch_tests'
+
+module Inferno
+  module Sequence
+    class OncStandalonePublicLaunchSequence < SequenceBase
+      include Inferno::Sequence::SharedONCLaunchTests
+
+      title 'Public Client Standalone Launch Sequence'
+
+      description 'Register Inferno as a public client with patient access and execute standalone launch.'
+
+      test_id_prefix 'OSLSP'
+
+      requires :onc_public_client_id,
+               :onc_public_scopes,
+               :onc_sl_oauth_authorize_endpoint,
+               :onc_sl_oauth_token_endpoint,
```

Comment:
Is there any reason this sequence uses `onc_sl_oauth_token_endpoint` (defined by `onc_standalone token_refresh_sequence`) while `onc_standalone_launch_sequence` uses `oauth_token_endpoint` defined by `onc_smart_dsicovery_sequence`?

`oauth_token_endpoint` is nice because you can keep testing even if the token refresh fails and they should be the same anyways.

https://github.com/onc-healthit/inferno-program/blob/6ec57461a2e9d490f36965261c47a3fe14109053/lib/modules/onc_program/onc_standalone_token_refresh_sequence.rb#L62

Reply:
This is where things get difficult because I can't chain a discovery sequence in front of this or the token revocation sequences.

EHR launches could use a different auth server.  As we run through the normal flow of the app, the auth server picked up in the standalone launch discovery will get overwritten by the auth server in the ehr launch discovery.  Then we will prefill with a wrong auth server.  This is a bigger deal on token revocation, where we need to be revoking the standalone patient token and not the ehr token, but by default when walking through we will prefill with the ehr auth server.

So, just to make this work, the standalone discovery launch steps saves the auth server 'off to the side' as a general record that the token revocation sequence and this sequence can use.